File: src/java/src/com/facebook/csslayout/CSSStyle.java
/**
 * Copyright (c) 2014, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */
package com.facebook.csslayout;

/**
 * The CSS style definition for a {@link CSSNode}.
 */
public class CSSStyle {

  public static final int SPACING_TOP = 0;
  public static final int SPACING_RIGHT = 1;
  public static final int SPACING_BOTTOM = 2;
  public static final int SPACING_LEFT = 3;

  public CSSFlexDirection flexDirection = CSSFlexDirection.COLUMN;
  public CSSJustify justifyContent = CSSJustify.FLEX_START;
  public CSSAlign alignItems = CSSAlign.STRETCH;
  public CSSAlign alignSelf = CSSAlign.AUTO;
  public CSSPositionType positionType = CSSPositionType.RELATIVE;
  public CSSWrap flexWrap = CSSWrap.NOWRAP;
  public float flex;

  public float[] margin = new float[4];
  public float[] padding = new float[4];
  public float[] border = new float[4];

  public float positionTop = CSSConstants.UNDEFINED;
  public float positionBottom = CSSConstants.UNDEFINED;
  public float positionLeft = CSSConstants.UNDEFINED;
  public float positionRight = CSSConstants.UNDEFINED;

  public float width = CSSConstants.UNDEFINED;
  public float height = CSSConstants.UNDEFINED;
}

File: src/csharp/Facebook.CSSLayout/CSSStyle.cs
/**
 * Copyright (c) 2014, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

namespace Facebook.CSSLayout
{
    /**
     * The CSS style definition for a {@link CSSNode}.
     */
    sealed class CSSStyle
    {
        public CSSDirection direction = CSSDirection.Inherit;
        public CSSFlexDirection flexDirection = CSSFlexDirection.Column;
        public CSSJustify justifyContent = CSSJustify.FlexStart;
        public CSSAlign alignContent = CSSAlign.FlexStart;
        public CSSAlign alignItems = CSSAlign.Stretch;
        public CSSAlign alignSelf = CSSAlign.Auto;
        public CSSPositionType positionType = CSSPositionType.Relative;
        public CSSWrap flexWrap = CSSWrap.NoWrap;
        public float flex;

        public Spacing margin = new Spacing();
        public Spacing padding = new Spacing();
        public Spacing border = new Spacing();

        public float[] position = {
            CSSConstants.Undefined,
            CSSConstants.Undefined,
            CSSConstants.Undefined,
            CSSConstants.Undefined
        };

        public float[] dimensions = {
            CSSConstants.Undefined,
            CSSConstants.Undefined
        };

        public float minWidth = CSSConstants.Undefined;
        public float minHeight = CSSConstants.Undefined;

        public float maxWidth = CSSConstants.Undefined;
        public float maxHeight = CSSConstants.Undefined;
    }
}
